code/datums/uplink/uplinks.dm

```diff
@@ -104,52 +106,30 @@ GLOBAL_LIST_EMPTY(world_uplinks)
 				var/path = UI.refund_path || UI.item
 				var/cost = UI.refund_amount || UI.cost
 				if(I.type == path && UI.refundable && I.check_uplink_validity())
-					uses += cost
+					crystals += cost
 					used_TC -= cost
 					to_chat(user, "<span class='notice'>[I] refunded.</span>")
 					qdel(I)
 					return
 		// If we are here, we didnt refund
 		to_chat(user, "<span class='warning'>[I] is not refundable.</span>")
 
-// HIDDEN UPLINK - Can be stored in anything but the host item has to have a trigger for it.
-/* How to create an uplink in 3 easy steps!
-
- 1. All obj/item 's have a hidden_uplink var. By default it's null. Give the item one with "new(src)", it must be in it's contents. Feel free to add "uses".
-
- 2. Code in the triggers. Use check_trigger for this, I recommend closing the item's menu with "usr << browse(null, "window=windowname") if it returns true.
- The var/value is the value that will be compared with the var/target. If they are equal it will activate the menu.
-
- 3. If you want the menu to stay until the users locks his uplink, add an active_uplink_check(mob/user as mob) in your interact/attack_hand proc.
- Then check if it's true, if true return. This will stop the normal menu appearing and will instead show the uplink menu.
-*/
-
-/obj/item/uplink/hidden
-	name = "hidden uplink"
-	desc = "There is something wrong if you're examining this."
-	var/active = 0
 
-// The hidden uplink MUST be inside an obj/item's contents.
-/obj/item/uplink/hidden/New()
-	spawn(2)
-		if(!istype(src.loc, /obj/item))
-			qdel(src)
-	..()
 
 // Toggles the uplink on and off. Normally this will bypass the item's normal functions and go to the uplink menu, if activated.
-/obj/item/uplink/hidden/proc/toggle()
+/datum/uplink/proc/toggle()
 	active = !active
 
 // Directly trigger the uplink. Turn on if it isn't already.
-/obj/item/uplink/hidden/proc/trigger(mob/user as mob)
+/datum/uplink/proc/trigger(mob/user as mob)
```

Comment:
`as mob` 😠

Reply:
Thanks, I only gave these a cursory glance while changing the path, ill tweak that now